cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ApiCredentials.java

```diff
@@ -0,0 +1,323 @@
+    private final CruiseControlSpec specSection;
+    private final OwnerReference ownerReference;
+
+    /**
+     * Constructs the Api Credentials Model for managing API users for Cruise Control API.
+     *
+     * @param namespace         Namespace of the cluster
+     * @param cluster           Name of the cluster to which this component belongs
+     * @param labels            Labels for Cruise Control instance
+     * @param ownerReference    Owner reference for Cruise Control instance
+     * @param specSection       Custom resource section configuring Cruise Control API users
+     */
+    public ApiCredentials(String namespace, String cluster, Labels labels, OwnerReference ownerReference, CruiseControlSpec specSection) {
+        this.namespace = namespace;
+        this.cluster = cluster;
+        this.labels = labels;
+        this.ownerReference = ownerReference;
+        this.specSection = specSection;
+
+        if (validateApiUsersConfig(specSection)) {
+            userManagedApiSecretName = specSection.getApiUsers().getValueFrom().getSecretKeyRef().getName();
+            userManagedApiSecretKey = specSection.getApiUsers().getValueFrom().getSecretKeyRef().getKey();
+        } else {
+            userManagedApiSecretName = null;
+            userManagedApiSecretKey = null;
+        }
+    }
+
+    /**
+     * @return  Returns user-managed API credentials secret name
+     */
+    public String getUserManagedApiSecretName() {
+        return userManagedApiSecretName;
+    }
+
+    /**
+     * @return  Returns user-managed API credentials secret key
+     */
+    /* test */ public String getUserManagedApiSecretKey() {
+        return userManagedApiSecretKey;
+    }
+
+    /**
+     * Checks if Cruise Control spec has valid ApiUsers config.
+     *
+     * @param ccSpec The Cruise Control spec to check.
+     */
+    private static boolean validateApiUsersConfig(CruiseControlSpec ccSpec)  {
+        HashLoginServiceApiUsers apiUsers = ccSpec.getApiUsers();
+        if (apiUsers != null)    {
+            if (apiUsers.getValueFrom() == null
+                    || apiUsers.getValueFrom().getSecretKeyRef() == null
+                    || apiUsers.getValueFrom().getSecretKeyRef().getName() == null
+                    || apiUsers.getValueFrom().getSecretKeyRef().getKey() == null) {
+                throw new InvalidResourceException("The configuration of the Cruise Control REST API users " +
+                        "referenced in spec.cruiseControl.apiUsers is invalid.");
+            } else {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Parse String containing API credential config into map of API user entries.
+     *
+     * @param config API credential config file as a String.
+     *
+     * @return map of API credential entries containing username, password, and role.
+     */
+    /* test */ public static Map<String, UserEntry> parseEntriesFromString(String config) {
```

Comment:
Same as above ... and there are more below. If you need to use it in tests, that is fine. But the test using the non-public / low-level methods like these should be in the same package. So you normally need only instead of making it public:

```suggestion
    /* test */ static Map<String, UserEntry> parseEntriesFromString(String config) {
```